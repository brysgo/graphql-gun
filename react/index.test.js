/* global describe, it, expect */
const gql = require("graphql-tag");
const Gun = require("gun/gun");
const React = require("react");
const gun = Gun();
const { createContainer, graphqlGun } = require("./")({ React, gun });

const renderer = require("react-test-renderer");

const Color = props => {
  return React.createElement(
    "div",
    { style: { color: props.color } },
    JSON.stringify(props, null, 2)
  );
};

describe("react", () => {
  describe("createContainer", () => {
    it("renders the fragment in the container", async () => {
      const chain = gun.get("rcCrtfitc");
      chain
        .get("color")
        .put({ hue: 255, saturation: 255, value: 255, other: "foo" });

      const ColorContainer = createContainer(Color, {
        fragments: {
          palette: gql`{
            rcCrtfitc {
              color {
                hue
                saturation
                value
              }
            }
          }`
        }
      });

      let containerInstance;
      const colorContainerElement = React.createElement(ColorContainer, {
        color: "blue",
        ref: container => containerInstance = container
      });
      const component = renderer.create(colorContainerElement);

      await containerInstance.promise;
      await containerInstance.promise;

      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it("reloads your component when new data comes", async () => {
      const chain = gun.get("rcCrycwndc");
      chain.get("thing").put({
        stuff: {
          one: "hello",
          two: "world"
        }
      });

      const ColorContainer = createContainer(Color, {
        fragments: {
          palette: gql`{
            rcCrycwndc {
              thing {
                stuff @live {
                  one
                  two
                }
              }
            }
          }`
        }
      });

      let containerInstance;
      const colorContainerElement = React.createElement(ColorContainer, {
        color: "red",
        ref: container => containerInstance = container
      });
      const component = renderer.create(colorContainerElement);

      const next = await containerInstance.promise;

      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();

      chain.get("thing").put({ stuff: { one: "new" } });

      await next();

      tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe("graphqlGun", () => {
    it("provides apollo client syntax for container", async () => {
      gun
        .get("color")
        .put({ hue: 255, saturation: 255, value: 255, other: "foo" });

      const ColorContainer = graphqlGun(
        gql`{
            color {
              hue
              saturation
              value
            }
          }`
      )(Color);

      let containerInstance;
      const colorContainerElement = React.createElement(ColorContainer, {
        color: "blue",
        ref: container => containerInstance = container
      });
      const component = renderer.create(colorContainerElement);

      await containerInstance.promise;

      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
