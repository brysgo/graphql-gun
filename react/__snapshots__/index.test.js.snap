// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react createContainer reloads your component when new data comes 1`] = `
<div
  style={
    Object {
      "color": "red",
    }
  }
>
  {
  "palette": {
    "thing": {
      "stuff": {
        "one": "hello",
        "two": "world"
      }
    }
  },
  "color": "red"
}
</div>
`;

exports[`react createContainer reloads your component when new data comes 2`] = `
<div
  style={
    Object {
      "color": "red",
    }
  }
>
  {
    "palette": {
      "thing": {
        "stuff": {
          "one": "new",
          "two": "world"
        }
      }
    },
    "color": "red"
  }
</div>
`;

exports[`react createContainer renders the fragment in the container 1`] = `
<div
  style={
    Object {
      "color": "blue",
    }
  }
>
  {
    "palette": {
      "color": {
        "hue": 255,
        "saturation": 255,
        "value": 255
      }
    },
    "color": "blue"
  }
</div>
`;

exports[`react graphqlGun provides apollo client syntax for container 1`] = `
<div
  style={
    Object {
      "color": "blue",
    }
  }
>
  {
  "data": {
    "color": {
      "hue": 255,
      "saturation": 255,
      "value": 255
    }
  },
  "color": "blue"
}
</div>
`;
